targets/unity-v2/Testing/Tests/PersistentSockets/PubSubTest.cs

```diff
@@ -0,0 +1,103 @@
+#if ENABLE_PLAYFABPUBSUB_API
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayFab.UUnit
+{
+    public class PubSubTest : UUnitTestCase
+    {
+        static PlayFab.EventsModels.EntityKey _MyEntityKey = null;
+        static string _previousTitleId = null;
+
+        private PubSub pubSub;
+        private Entity entity;
+
+        private int testTicks = 0;
+        private int playstreamTickDelay = 150;
+
+        public override void SetUp(UUnitTestContext testContext)
+        {
+            // specific title id for relay test
+            _previousTitleId = PlayFabSettings.TitleId;
+            PlayFabSettings.TitleId = "70B02F89";
+            PlayFabSettings.VerticalName = "spi";
+        }
+
+        public override void Tick(UUnitTestContext testContext)
+        {
+            ++testTicks;
+            // this test will ping a write event continuously
+            // THIS TEST THROTTLES THE TITLE. so you should NOT do this until you KNOW the object is open
+            if (pubSub != null && pubSub.State == PersistentSocketState.Opened && testTicks % playstreamTickDelay == 0)
+            {
+                EventsModels.WriteEventsRequest req = new EventsModels.WriteEventsRequest();
+
+                EventsModels.EventContents ec = new EventsModels.EventContents();
+
+                ec.Entity = new EventsModels.EntityKey();
+                ec.Entity.Id = _MyEntityKey.Id;
+                ec.Entity.Type = _MyEntityKey.Type;
+                ec.Name = "testevent";
+
+                ec.EventNamespace = "com.playfab.events.test";
+
+                req.Events = new List<EventsModels.EventContents>();
+                req.Events.Add(ec);
+
+                PlayFabEventsAPI.WriteEvents(req, WriteEventSuccessful, WriteEventFail);
+            }
+        }
+
+        public void WriteEventSuccessful(EventsModels.WriteEventsResponse response)
+        {
+            // check that we have assigned id's?
+            // this will probably get called a few times before the hub connection opens
+            // but once the hub connection opens it should immediatley start seeing something each Tick
+        }
+
+        public void WriteEventFail(PlayFabError error)
+        {
+            Debug.Log("Write Event Failed: " + error.ErrorMessage);
```

Comment:
Is a WriteEvent failure a reason to fail the test?
I'd typically prefer a test failure over a Debug Log.
Generally I don't want tests to debug log at all.